nucypher/blockchain/eth/actors.py

```diff
@@ -211,22 +211,29 @@ def deploy_contract(self,
                         contract_name: str,
                         gas_limit: int = None,
                         plaintext_secret: str = None,
+                        bare: bool = False,
                         progress=None,
-                        *args,
-                        **kwargs,
+                        *args, **kwargs,
                         ) -> Tuple[dict, BaseContractDeployer]:
 
         Deployer = self.__get_deployer(contract_name=contract_name)
         deployer = Deployer(registry=self.registry,
                             deployer_address=self.deployer_address,
                             economics=self.economics,
-                            *args,
-                            **kwargs)
+                            *args, **kwargs)
+
         if Deployer._upgradeable:
-            if not plaintext_secret:
-                raise ValueError("Upgrade plaintext_secret must be passed to deploy an upgradeable contract.")
-            secret_hash = keccak(bytes(plaintext_secret, encoding='utf-8'))
-            txhashes = deployer.deploy(secret_hash=secret_hash, gas_limit=gas_limit, progress=progress)
+            is_initial_deployment = not bare
+            if is_initial_deployment and not plaintext_secret:
+                raise ValueError("An upgrade secret must be passed to perform an initial deployment series"
```

Comment:
I am trying to indicate that this action will result in a series of transactions, since it is the *initial deployment* it also will deploy the proxy contract, set it's target, then perform any contract-specific initialization steps.

Reply:
Ok. What about something like: `An upgrade secret must be passed to perform all required initial deployment operations`?

I didn't provide as a suggestion since the existing code uses multiple lines.

Reply:
Ok. What about something like: `An upgrade secret must be passed to perform all required initial deployment operations?`

I didn't provide as a suggestion since the existing code uses multiple lines.

Reply:
:heavy_check_mark: